server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -742,24 +911,19 @@ public void setFileFlushPolicy(TsFileFlushPolicy fileFlushPolicy) {
 
   public boolean isFileAlreadyExist(TsFileResource tsFileResource, PartialPath storageGroup,
       long partitionNum) {
-    StorageGroupProcessor processor = processorMap.get(storageGroup);
-    return processor != null && processor.isFileAlreadyExist(tsFileResource, partitionNum);
-  }
-
-  public static long getTimePartitionInterval() {
-    if (timePartitionInterval == -1) {
-      initTimePartition();
+    VirtualStorageGroupManager virtualStorageGroupManager = processorMap.get(storageGroup);
+    if (virtualStorageGroupManager == null) {
+      return false;
     }
-    return timePartitionInterval;
-  }
-
-  @TestOnly
-  public static void setTimePartitionInterval(long timePartitionInterval) {
-    StorageEngine.timePartitionInterval = timePartitionInterval;
-  }
 
-  public static long getTimePartition(long time) {
-    return enablePartition ? time / timePartitionInterval : 0;
+    for (StorageGroupProcessor storageGroupProcessor : virtualStorageGroupManager
+        .getAllVirutalStorageGroupProcessor()) {
+      if (storageGroupProcessor != null && storageGroupProcessor
+          .isFileAlreadyExist(tsFileResource, partitionNum)) {
+        return true;
+      }
+    }
```

Comment:
This is a bit inefficient, and probably you can infer the virtual storage group of the file using its devices.

Reply:
fixed~